autorest/client.go

```diff
@@ -43,6 +28,12 @@ const (
 `
 )
 
+// Response serves as the base for all responses from generated clients. It provides access to the
+// last http.Response.
+type Response struct {
```

Comment:
This makes me think do we really need an `autorest.Response`? If you're defining certain methods on it, you can often just do:

    type Response *http.Response

instead of embedding it.